### DISCUSSION TITLE
Qual a melhor forma de aprender JavaScript?

### DISCUSSION THREAD

**Discussion by Guilherme-Santos08** [frontendbr/forum#1965] [Perguntas e Respostas]
Sempre fico em dúvida sobre aprender JavaScript com cursos ou aprender através de artigos!! Poderiam me ajudar?

**Comment by brect** [comment]
Documentação

**Reply by memachado** [reply]
Complementaria:

- Estuda matemática básica
- Estuda lógica básica
- Estuda lógica de programação

Depois quase todas as linguagens de programação são suficientemente explicadas na documentação.

**Reply by Guilherme-Santos08** [reply]
Obrigado!!!

**Comment by theandersonn** [comment]
Guilherme não existe um segredo de ouro, uma fórmula infalível. É normal alguns se adaptarem com determinado material e outros não. E nisso podemos aplicar para qualquer aprendizado.

Um aspecto legal é que temos muito conteúdo disponível.  

No meu caso eu consumo material em vídeo e artigos. Aplico o que aprendo nas minhas tarefas do trabalho, e quando a demanda não requer aquele conhecimento, faço um projeto fictício.

Resolver problemas reais nos ajuda a evoluir. Uma coisa é completarmos um curso ou tutorial na qual a solução é guiada, e outra é lidarmos com problemas reais, mesmo em projetos fictícios. Esse processo nos desenvolve em pesquisas e implementações técnicas.

Gosto de receber feedbacks das minhas implementações, saber o que posso fazer melhor, saber se o código que escrevi está legível e etc... Mas há um tempo atrás ficava apavorado em mostrar algo que fiz, não tinha confiança.  Esse é outro detalhe, temos que ter paciência conosco. Se o estudo for regular, a evolução acontecerá.

**Reply by Guilherme-Santos08** [reply]
Muito obrigado por compartilhar sua opinião , me ajudo bastante!!! Acredito que eu aprendo mais com artigos, muitas vezes o curso/tutorial fica cansativo e eu  perco o foco.

**Comment by wilsonneto-swe** [comment]
Vou deixar aqui minha opinião também, foi como eu aprendi, e como eu estudo até hoje... Mesmo eu trabalhando já com ReactJS há alguns anos (trabalhando com programação em geral há 10 anos) e atualmente também com NextJS, eu faço muitos cursos ainda, faço para aprender detalhes e pegar insights com desenvolvedores que dominam tanto ou/e mais que eu... Leio e lia no inicio muitos artigos diariamente, para me manter informado, assisto e assistia a muitos vídeos, leio e lia muita, mas muita documentação... 

E acredito que o que mais vai te fazer realmente evoluir, e foi o que me fez evoluir, é praticar, fazer seus projetos, ser curioso além do curso e fazer as suas POCs e projetos, do zero, sem apenas copiar códigos de cursos, tutoriais ou artigos.

Resumindo, não acho que seja questão de aprender de um jeito ou de outro, aprenda de todos os jeitos, e invista mais tempo praticando do que o tempo que você investe estudando, ou assistindo vídeos, ou cursos ou artigos... 

Foi assim comigo, espero que ajude, forte abraço!

**Reply by Guilherme-Santos08** [reply]
Obrigado por compartilhar sua experiência, wilson!!! Vai me ajudar bastante!!!
